Marketplace/guitar/guitar.js

```diff
@@ -0,0 +1,666 @@
+                sounds["Blue" + i + "R"] = SoundCache.getSound(Script.resolvePath("sounds/blue/" + i + "R.wav"));
+            }
+        },
+        mousePress: function(id, event) {
+            if (overlays.indexOf(id) !== NOT_FOUND) {
+                if (timeout) {
+                    Script.clearTimeout(timeout);
+                    timeout = null;
+                }
+                timeout = Script.setTimeout(function(){
+                    _this.releaseGrab();
+                    timeout = null;
+                }, STILL_PLAYING_TIMEOUT_MS);
+                if (ampCheckInterval) {
+                    Script.clearTimeout(ampCheckInterval);
+                }
+                ampCheckInterval = Script.setInterval(this.updateAmpPositions, UPDATE_POSITION_MS);
+                var name = Overlays.getProperty(id, 'name');
+                if (name.indexOf("Neck") !== -1) {
+                    neckGrabColor = name.substr(NECK_COLOR_INDEX);
+                } else {
+                    var currentSoundL = sounds[neckGrabColor + name.substr(RIFF_NUMBER_INDEX) + "L"];
+                    var currentSoundR = sounds[neckGrabColor + name.substr(RIFF_NUMBER_INDEX) + "R"];
+                    if (currentSoundL.downloaded) {
+                        if (injectorL) {
+                            injectorL.stop();
+                        }
+                        injectorL = Audio.playSound(currentSoundL, {
+                            position: ampLeftPosition,
+                            volume: AUDIO_VOLUME_LEVEL,
+                            loop: true
+                        });
+                    }
+                    if (currentSoundR.downloaded) {
+                        if (injectorR) {
+                            injectorR.stop();
+                        }
+                        injectorR = Audio.playSound(currentSoundR, {
+                            position: ampRightPosition,
+                            volume: AUDIO_VOLUME_LEVEL,
+                            loop: true
+                        });
+                    }
+                }
+            }
+        },
+        setNeckColor: function(buttonID, params) {
+            neckGrabColor = params[0];
+        },
+        startNearGrab: function() {
+            if (!colorsShowing) {
+                this.createFingertipEntity("LeftHandMiddle4");
+                this.createFingertipEntity("RightHandMiddle4");
+                if (interval) {
+                    Script.clearInterval(interval);
+                }
+                interval = Script.setInterval(this.updatePositions, POSITION_CHECK_INTERVAL_MS);
+                Script.setTimeout(function () {
+                    Script.clearInterval(interval);
+                }, POSITION_CHECK_TIMEOUT_MS);
+                this.createBodyOverlays();
+                this.createNeckOverlays();
+                numberHandsGrabbing = ONE_HAND;
```

Comment:
I think trying to predict this is going to end up in strange states - especially because you're not technically "grabbing" with the second hand when you're playing.